backends/bmv2/JsonObjects.cpp

```diff
@@ -118,8 +118,14 @@ JsonObjects::add_meta_info() {
  */
 unsigned
 JsonObjects::add_header_type(const cstring& name, Util::JsonArray*& fields, unsigned max_length) {
+    std::string sname(name, name.size());
+    try {
+        return header_type_id.at(sname);
+    } catch (const std::out_of_range &x) {
```

Comment:
Using exceptions for control flow is not a good practice. I recommend using the following pattern:
```
auto header_type_id_it = header_type_id.find(sname);
if( header_type_id_it != header_type_id.end() )
  return header_type_id_it->second;
}
```